client/driver/exec.go

```diff
@@ -92,6 +92,13 @@ func (d *ExecDriver) Periodic() (bool, time.Duration) {
 	return true, 15 * time.Second
 }
 
+func (d *ExecDriver) Prestart(execctx *ExecContext, task *structs.Task) error {
+	// Set the host environment variables.
+	filter := strings.Split(d.config.ReadDefault("env.blacklist", config.DefaultEnvBlacklist), ",")
```

Comment:
I (indirectly) added it to TaskRunner by putting it in the GetTaskEnv factory: https://github.com/hashicorp/nomad/pull/2054/commits/aaa70ab7b80645de41a9465198fde2cbae69751d